proposed/2022/cpm-dotnet-cli-add-package-support.md

```diff
@@ -4,218 +4,146 @@
-| 5 | ✔️ | ✔️ | ✔️ | ✔️ | **dotnet CLI** - Restore failed with NU1008 error and NO edits were made to the csproj file. **VS** - Clicked on Update package in PM UI. New `PackageVersion` added to csproj file and `VersionOverride` in .csproj file was not updated | Update `VersionOverride` attribute value in the corresponding `PackageReference` item in .(cs/vb)proj file. | ✔️ |
-
-> [!NOTE]
-> Scenarios with multiple Directory.Packages.props are out of scope for now.
-
-### 1. The package reference does not exist
-
-If the package does not already exist, it should be added along with the appropriate package version to `Directory.packages.props`. The package version should either be the latest version or the one specified in the CLI command. Only the package name (not the version) should be added to `<PackageReference>` in the project file.
-
-#### Before `add package` is executed
-
-The props file:
-
-```xml
-<Project>
-    <PropertyGroup>
-        <ManagePackageVersionsCentrally>true</ManagePackageVersionsCentrally>
-    </PropertyGroup>
-    <ItemGroup>
-    </ItemGroup>
-</Project>
-```
-
-The .csproj file:
-
-```xml
-<Project Sdk="Microsoft.NET.Sdk">
-    <PropertyGroup>
-        <TargetFramework>net6.0</TargetFramework>
-        <ImplicitUsings>enable</ImplicitUsings>
-        <Nullable>enable</Nullable>
-    </PropertyGroup>
-
-    <ItemGroup>
-    </ItemGroup>
-</Project>
-```
-
-#### After `add package` is executed
-
-`dotnet add ToDo.csproj package Newtonsoft.Json -v 13.0.1`
-
-The props file:
-
-```xml
-<Project>
-    <PropertyGroup>
-        <ManagePackageVersionsCentrally>true</ManagePackageVersionsCentrally>
-    </PropertyGroup>
-    <ItemGroup>
-    <PackageVersion Include="Newtonsoft.Json" Version="13.0.1"/>
-    </ItemGroup>
-</Project>
-```
-
-The .csproj file:
-
-```xml
-<Project Sdk="Microsoft.NET.Sdk">
-    <PropertyGroup>
-        <TargetFramework>net6.0</TargetFramework>
-        <ImplicitUsings>enable</ImplicitUsings>
-        <Nullable>enable</Nullable>
-    </PropertyGroup>
-
-    <ItemGroup>
-        <PackageReference Include="Newtonsoft.Json"/>
-    </ItemGroup>
-</Project>
-```
-
-In case there are multiple `Directory.packages.props` files in the repo, the props file that is closest must be considered.
-
-```xml
-Repository
- |-- Directory.Packages.props
- |-- Solution1
+| Scenario # | PackageReference exists? | VersionOverride exists? | PackageVersion exists? | Is Version passed from the commandline? | New behavior in dotnet CLI | In Scope |
```

Comment:
The e-mail says only the first 4 scenarios are in scope, but this doc still selects more. 
Maybe make t hat more apparent in some way?

Reply:
What I am thinking is that the first 4 scenarios are in scope for @pragnya17's work. The implementation for the first project will be merged into a feature branch https://github.com/NuGet/NuGet.Client/pull/4700. I was wondering if we can address other scenarios and do a single merge into `dev` branch covering all the edge cases listed in the spec. Thoughts?

Reply:
Ah, I probably missed that. 

I think some of the scenarios not handled are more important than others. 

Skimming through the scenarios 5-16, i think I can live without some getting implemented in V1 and others even not being implemented at all. 

On the other hand, I'd love for us to deliver value to the customers. 

Maybe rather than gating that feature branch getting merged to dev on the other scenarios getting addressed, maybe we prioritize those and figure out how many of those we feel like are critical to the experience altogether. 

I think it'd be great if we can have this in 7.0.100.

Reply:
I had a quick chat with @aortiz-msft about this and I am happy to address scenarios 9 to 16 in the upcoming quality week this sprint (July 2022).  Most of them are either no-op or error and I guess it should be quick. Once all the scenarios are addressed, we can merge that into dev branch. Hope you are okay with this proposal.

Reply:
Love it! 

Thanks @kartheekp-ms